Example/Core/Tests/FIRAppTest.m

```diff
@@ -735,6 +736,19 @@ - (void)testMultipleLibraries {
   XCTAssertTrue([[FIRApp firebaseUserAgent] containsString:@"LegalName/1.0.0 LegalName2/2.0.0"]);
 }
 
+- (void)testRegisteringConformingLibrary {
+  Class testClass = [FIRTestClass class];
+  [FIRApp registerInternalLibrary:testClass withName:@"LegalName" withVersion:@"1.0.0"];
+  XCTAssertTrue([[FIRApp firebaseUserAgent] containsString:@"LegalName/1.0.0"]);
+}
+
+- (void)testRegisteringNonConformingLibrary {
+  XCTAssertThrows([FIRApp registerInternalLibrary:[NSString class]
+                                         withName:@"IllegalLibrary"
+                                      withVersion:@"1.0.0"]);
+  XCTAssertFalse([[FIRApp firebaseUserAgent] containsString:@"LegalName/1.0.0"]);
```

Comment:
Nit: Should this be checking `containsString:@"IllegalLibrary/1.0.0"`?

Reply:
Good find! Fixed.